SKIRT/core/SecondarySourceSystem.cpp

```diff
@@ -421,40 +421,76 @@ namespace
         {
             // first generate a random azimuth angle
             const double phi = 2. * M_PI * random->uniform();
-            // compute its sine and cosine
-            const double cosphi = cos(phi);
-            const double sinphi = sin(phi);
 
             // now generate a random zenith angle
             // first, compute the cumulative distribution for a photon at the given wavelength
-            // create an empty array with the same size as the zenith angle grid
+            // create empty PDF and CDF arrays with the same size as the zenith angle grid
             const Array& thetas = _ms->mix(_m, _hv[0])->thetaGrid();
-            Array cdf(thetas.size());
+            Array pdf(thetas.size()), cdf(thetas.size());
+            ;
```

Comment:
This semicolon can/should be removed?

Reply:
Yes, no idea how that ended up there. Has been removed.